hs-abci-sdk/src/Tendermint/SDK/Application/AnteHandler.hs

```diff
@@ -31,32 +31,49 @@ applyAnteHandler
   -> TransactionApplication (Sem r)
 applyAnteHandler (AnteHandler ah) = ($) ah
 
+createAccountAnteHandler
+  :: Members A.AuthEffs r
+  => AnteHandler r
+createAccountAnteHandler = AnteHandler $
+  \txApplication tx@(RoutingTx Tx{..}) -> do
+    let Msg{msgAuthor} = txMsg
+    mAcnt <- A.getAccount msgAuthor
+    case mAcnt of
+      Nothing -> void $ A.createAccount msgAuthor
+      _       -> pure ()
+    txApplication tx >>= pure
+
 nonceAnteHandler
   :: Members A.AuthEffs r
   => Member (Error AppError) r
   => AnteHandler r
 nonceAnteHandler = AnteHandler $
   \txApplication tx@(RoutingTx Tx{..}) -> do
     let Msg{msgAuthor} = txMsg
-    mAcnt <- A.getAccount msgAuthor
-    account <- case mAcnt of
-      Just a@A.Account{accountNonce} -> do
-        unless (accountNonce <= txNonce) $
-          throwSDKError (NonceException accountNonce txNonce)
-        pure a
+    preMAcnt <- A.getAccount msgAuthor
+    case preMAcnt of
+      Just A.Account{accountNonce} -> do
+        let expectedNonce = accountNonce + 1
+        unless (txNonce == expectedNonce) $
+          throwSDKError (NonceException expectedNonce txNonce)
       Nothing -> do
-        unless (txNonce == 0) $
-          throwSDKError (NonceException 0 txNonce)
-        A.createAccount msgAuthor
+        unless (txNonce == 1) $
+          throwSDKError (NonceException 1 txNonce)
     result <- txApplication tx
-    A.putAccount msgAuthor $
-      account { A.accountNonce = A.accountNonce account + 1}
+    postMAcnt <- A.getAccount msgAuthor
+    case postMAcnt of
+      Just acnt@A.Account{accountNonce} -> do
+        A.putAccount msgAuthor $
+          acnt { A.accountNonce = accountNonce + 1}
+      -- @NOTE: no-op when no nonce is availble to update
+      Nothing -> pure ()
     pure result
 
 baseAppAnteHandler
   :: Members A.AuthEffs r
   => Member (Error AppError) r
   => AnteHandler r
 baseAppAnteHandler = mconcat $
-  [ nonceAnteHandler
```

Comment:
maybe make a comment about how these are run in order starting from the first in the list